core/src/main/java/com/linecorp/armeria/common/util/SystemInfo.java

```diff
@@ -109,6 +111,12 @@
                 JETTY_ALPN_OPTIONAL_OR_AVAILABLE = false;
             }
         }
+
+        try {
+            osType = OsType.valueOf(Ascii.toUpperCase(System.getProperty("os.name", "").split(" ")[0]));
```

Comment:
OS detection is a bit more complicated unfortunately.

Can you use the if statements for the `OsType` OS's from this famous OS detection logic?

https://github.com/trustin/os-maven-plugin/blob/master/src/main/java/kr/motd/maven/os/Detector.java#L150

Reply:
OK. 😃 Changed to an if statement.